cloudbuild/python-scripts/parse_logs.py

```diff
@@ -0,0 +1,513 @@
+        # If for any of the record, filter condition is not met then throw an error.
+        is_query_result_present = check_query_correctness(
+            gcs_log_object, logs_as_string
+        )
+
+    if metric_string in logs_as_string:
+        # If metric string is present in the logs, extract the value.
+        metric_value = extract_metric(
+            logs_as_string, metric_string, end_of_metric_string
+        )
+        return metric_value, is_query_result_present
+    # If not return -1
+    return -1, is_query_result_present
+
+
+def get_logs_pattern(client_project_name, region, job_id):
+    """Method to return dataproc job state and yarn logs path.
+
+    Args:
+        client_project_name: Project ID of the GCP project that contains the
+          cluster.
+        region: region in which the cluster runs.
+        job_id: JOB_Id of the dataproc job.
+
+    Returns:
+        The path pattern inside the temp bucket in which logs are found.
+
+    Raises:
+        RuntimeError: In case the Dataproc job fails
+    """
+    # Create a client
+    client = dataproc_v1.JobControllerClient(
+        client_options={'api_endpoint': f'{region}-dataproc.googleapis.com:443'}
+    )
+
+    # Initialize request argument(s)
+    request = dataproc_v1.GetJobRequest(
+        project_id=client_project_name,
+        region=region,
+        job_id=job_id,
+    )
+
+    # Make the request
+    response = client.get_job(request=request)
+    state = response.status.state.name
+
+    if state == 'ERROR':
+        raise RuntimeError(
+            f'[Log: parse_logs ERROR] Dataproc Job with JOB ID: "{job_id}" failed'
+        )
+
+    # If the dataproc job did not fail, continue to match the number of records.
+    cluster_id = response.placement.cluster_uuid
+    tracking_url = response.yarn_applications[0].tracking_url
+
+    # Tracking url is of the form http://.../proxy/application_..._.../
+    # We need to extract the application number which is at the end of this url.
+    # The pattern
+    #     /([^/]+): Searches for a '/'. This is followed by set of characters
+    #     which are not a '/', as a group.
+    #     Optional '/' at the end (as some urls lack the ending '/')
+    # Thus the current pattern helps in searching 'application_YYYYYY_XXXX'
+    # as the part of group 1.
+
+    tracking_url_pattern = r'/([^/]+)(?:/)?$'
+    yarn_application_number = re.search(tracking_url_pattern, tracking_url).group(
+        1
+    )
+    # With the extracted application number of the format
+    # 'application_YYYYYY_XXXX'
+    # We need to extract the actual application number within the cluster,
+    # i.e. the ending digits which is the job number in the cluster.
+    # The pattern
+    #     [^_]+: Searches for the last occurrence of characters
+    #     which are not a "_".
+    #     i.e. the job number present after the last "_"
+    # Thus the current pattern helps in searching "XXXX" as the part of group 0.
+
```

Comment:
Please remove this line. Comments should be directly above concerned code.

Reply:
Fixed.